src/backend/common/helpers/insights_helper.py

```diff
@@ -898,6 +896,46 @@ def doOverallMatchInsights(self) -> List[Insight]:
 
         return insights
 
+    @classmethod
+    def _calculate_einstein_streaks(cls, division_winners_map: Dict[str, List[int]]) -> Dict[str, int]:
+
+        einstein_streak_output = defaultdict(int)
+        for team_key, unsorted_years_list in division_winners_map.items():
+            years = sorted(unsorted_years_list)
+
+            if not years:
+                einstein_streak_output[team_key] = 0
+                continue
+
+            # Initialize streaks
+            # Every team with at least one year has a max streak of at least 1
+            max_streak_for_team = 1
+            current_streak = 1
+            last_year_in_streak = years[0]
+
+            # Iterate starting from the second year
+            for i in range(1, len(years)):
+                current_year = years[i]
+                # A streak continues if current_year is last_year_in_streak + 1
+                # OR if last_year_in_streak was 2019 and current_year is 2022 (COVID gap)
+                is_consecutive = (current_year == last_year_in_streak + 1) or \
+                                 (last_year_in_streak == 2019 and current_year == 2022)
+
+                if is_consecutive:
+                    current_streak += 1
+                else:
+                    # Streak broken, current_streak for the new potential streak starts at 1
+                    current_streak = 1
+
+                if current_streak > max_streak_for_team:
+                    max_streak_for_team = current_streak
```

Comment:
nit: could do this with a `max(current_streak, max_streak_for_team)` to remove some flow control